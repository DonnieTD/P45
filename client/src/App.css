.App {
  text-align: center;
  background-color: black;
  min-height: 100%;
  height: 100%;
}

.navBar .navbar-brand{
  border-bottom: solid 1px limegreen;
  margin-bottom: 5vh;
  color:limegreen!important;
}

.formOne{
  width: 55vw;
  border-bottom: solid 1px limegreen;
  border-top: solid 1px limegreen;
  margin: 2.5vh auto;
}

.formTwo{
  width: 55vw;
  border-bottom: solid 1px limegreen;
  border-top: solid 1px limegreen;
  margin: 10vh auto 0vh auto;
}