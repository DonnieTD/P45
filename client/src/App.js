import React, { useState } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Row, Col, Form, Button, Navbar } from 'react-bootstrap';
import './App.css';
import axios from 'axios';

function App() {
  const [x, setX] = useState(0);
  const [y, setY] = useState(0);
  const [deploy, setDeploy] = useState("")
  const [outputText, setOutputText] = useState(['No Rovers Deployed']);
  function SubmitionHandler(e) {
    e.preventDefault();

    let dataArr = deploy.replace(/ /g, '').split(';').filter(n => n);

    if (dataArr.length % 2 == 0) {
      let AjaxData = {
        plane: {
          "x": x,
          "y": y
        },
        rovers: []
      };

      for (let i = 0; i < dataArr.length - 1; i += 2) {
        let arrWithoutNewLines = dataArr[i].replace(/[\r\n]+/gm, "");

        let rover = {
          "x": Number(arrWithoutNewLines[0]),
          "y": Number(arrWithoutNewLines[1]),
          "d": arrWithoutNewLines[2],
          "moves": dataArr[i + 1].replace(/[\r\n]+/gm, "")
        };

        AjaxData.rovers.push(rover);
      };

         axios({ method: 'post', url: 'http://localhost:3003/Rover', data: AjaxData })
        .then(function (response) {
          if (response.data.length >= 1) {
            let output = [];
            response.data.forEach((x, i) => {
              output.push("Rover " + (i + 1) + " Postition : {x :" + JSON.stringify(x.x) + ", y: " + JSON.stringify(x.y) + ", d: " + JSON.stringify(x.d) + " } ");
            })
            setOutputText(output);
          }
        })
        .catch(function (error) {
          setOutputText([error.response.data]);
        });
    } else {
      setOutputText(['Invalid Input Format Please Check Example']);
    }
  }

  return (
    <div className="App">
      <Navbar className="navBar">
        <Navbar.Brand href="#home">Rover</Navbar.Brand>
      </Navbar>
      <Row>
        <Col xs={12} md={12}>
          <h2>Output</h2>
          <Form className="formOne">
            {outputText.map((x,i) => { return <p key={i}style={{ marginBottom: '0px' }}>{x}</p> })}
          </Form>
          <Form className="formTwo" onSubmit={SubmitionHandler}>
            <h2 style={{ marginTop: '2.5vh' }}>Input</h2>
            <Form.Group controlId="exampleForm.ControlInput1" style={{ marginBottom: '1vh' }}>
              <Row>
                <Col md={6}>
                  <Form.Label>Plane x:</Form.Label>
                  <Form.Control onChange={e => setX(e.target.value)} type="number" placeholder="Eg: 5" />
                </Col>
                <Col md={6}>
                  <Form.Label>Plane y:</Form.Label>
                  <Form.Control onChange={e => setY(e.target.value)} type="number" placeholder="Eg: 5" />
                </Col>
                <Col style={{ marginTop: '5vh' }} md={12}>
                  <Form.Label>Deploy Instructions: (Please split input lines with semi-colons)</Form.Label>
                  <Form.Control onChange={e => setDeploy(e.target.value)} as="textarea" rows="3" />
                </Col>
                <Button type="submit" style={{ margin: '5vh auto', background: "white", color: 'black' }}>
                  Submit
                </Button>
              </Row>
            </Form.Group>
          </Form>

          <p style={{ color: 'white' }}>
            Example<br />
            1 2 N; LMLM;<br />
            or<br />
            1 2 N;
            LMLMLMLMM;
            3 3 E;
            MMRMMRMRRM;
              </p>
        </Col>
      </Row>
    </div>
  );
}

export default App;
