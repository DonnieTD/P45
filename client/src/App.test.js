import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

it('Snapshot of App', () => {
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
});