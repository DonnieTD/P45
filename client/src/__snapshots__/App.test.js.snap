// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of App 1`] = `
<div
  className="App"
>
  <nav
    className="navBar navbar navbar-expand navbar-light"
  >
    <a
      className="navbar-brand"
      href="#home"
    >
      Rover
    </a>
  </nav>
  <div
    className="row"
  >
    <div
      className="col-md-12 col-12"
    >
      <h2>
        Output
      </h2>
      <form
        className="formOne"
      >
        <p
          style={
            Object {
              "marginBottom": "0px",
            }
          }
        >
          No Rovers Deployed
        </p>
      </form>
      <form
        className="formTwo"
        onSubmit={[Function]}
      >
        <h2
          style={
            Object {
              "marginTop": "2.5vh",
            }
          }
        >
          Input
        </h2>
        <div
          className="form-group"
          style={
            Object {
              "marginBottom": "1vh",
            }
          }
        >
          <div
            className="row"
          >
            <div
              className="col-md-6"
            >
              <label
                className="form-label"
                htmlFor="exampleForm.ControlInput1"
              >
                Plane x:
              </label>
              <input
                className="form-control"
                id="exampleForm.ControlInput1"
                onChange={[Function]}
                placeholder="Eg: 5"
                type="number"
              />
            </div>
            <div
              className="col-md-6"
            >
              <label
                className="form-label"
                htmlFor="exampleForm.ControlInput1"
              >
                Plane y:
              </label>
              <input
                className="form-control"
                id="exampleForm.ControlInput1"
                onChange={[Function]}
                placeholder="Eg: 5"
                type="number"
              />
            </div>
            <div
              className="col-md-12"
              style={
                Object {
                  "marginTop": "5vh",
                }
              }
            >
              <label
                className="form-label"
                htmlFor="exampleForm.ControlInput1"
              >
                Deploy Instructions: (Please split input lines with semi-colons)
              </label>
              <textarea
                className="form-control"
                id="exampleForm.ControlInput1"
                onChange={[Function]}
                rows="3"
              />
            </div>
            <button
              className="btn btn-primary"
              disabled={false}
              style={
                Object {
                  "background": "white",
                  "color": "black",
                  "margin": "5vh auto",
                }
              }
              type="submit"
            >
              Submit
            </button>
          </div>
        </div>
      </form>
      <p
        style={
          Object {
            "color": "white",
          }
        }
      >
        Example
        <br />
        1 2 N; LMLM;
        <br />
        or
        <br />
        1 2 N; LMLMLMLMM; 3 3 E; MMRMMRMRRM;
      </p>
    </div>
  </div>
</div>
`;
